sklearn/ensemble/tests/test_forest.py

```diff
@@ -204,6 +204,18 @@ def check_probability(name):
                                   np.exp(clf.predict_log_proba(iris.data)))
 
 
+def check_raises_n_estimator_warning(name):
+    ForestClassifier = FOREST_CLASSIFIERS[name]
+    clf = ForestClassifier(n_estimators=10, random_state=1, max_features=1,
+                           max_depth=1)
+    assert_warns(DeprecationWarning, func=clf.fit, X=iris.data, y=iris.target)
+
+
+@pytest.mark.parametrize('name', FOREST_ESTIMATORS)
+def test_n_estimator_raises_warning(name):
+    check_raises_n_estimator_warning(name)
```

Comment:
There is no need to have a separate function.

Please also test that no warning is raised if the user explicitly passed a value